scripts/make-league.js

```diff
@@ -0,0 +1,153 @@
+const { League } = require('../models/league.js');
+
+const readline = require('readline').createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+const exec = require('child_process').execSync;
+const path = require('path');
+const os = require('os');
+const fs = require('fs').promises;
+
+async function query(message) {
+    return new Promise((resolve, _) => {
+        readline.question(message + ': ', (answer) => {
+            resolve(answer);
+        });
+    });
+}
+
+async function okPrompt() {
+    const response = await query("Is this ok? [y/n]");
+    switch (response) {
+        case 'y':
+        case 'Y':
+            return true;
+        case 'n':
+        case 'N':
+            return false;
+        default:
+            console.log(`I don't understand, try again`);
+            return await okPrompt();
+    }
+}
+
+/**
+ * Interrupts current script,
+ * opens file in $EDITOR,
+ * waits for user to exit $EDITOR,
+ * then resumes current script
+ */
+function openFileInEditor(filename) {
+    const editor = process.env['EDITOR'];
+    const opts = {
+        stdio: 'inherit'
+    };
+    return exec(`${editor} ${filename}`, opts);
+}
+
+/**
+ * @param obj: 'a
+ * @param toText: 'a -> string
+ * @param fromText: string -> 'b
+ * @returns: 'b
+ */
+async function editObjectInEditor(obj, toText, fromText) {
+    const text = toText(obj);
+    const randomName = `tmp-${Math.floor(Math.random() * Math.floor(1000))}`;
+    const tmpFile = path.join(os.tmpdir(), randomName);
+    // create file
+    // write text to file
+    await fs.writeFile(tmpFile, text);
+    openFileInEditor(tmpFile);
+    // read file
+    const result = await fs.readFile(tmpFile, {encoding: 'utf-8'});
```

Comment:
Could clean the file here after editing is done?

Reply:
tmp dir gets cleared out on system reboot, so I think it's fine?